website/docs/terms/hover-terms.md

```diff
@@ -0,0 +1,136 @@
+
+grain:
+  displayText: grain  
+  hoverSnippet: Your data's grain is the combination of columns at which records in a table are unique. Ideally, this is captured in a single column and a unique primary key.
+
+idempotent:
+  displayText: idempotent
+  hoverSnippet: Idempotent describes a process that gives you the same result no matter how many times you run it.
+
+json:
+  displayText: JSON
+  hoverSnippet: JSON (JavaScript Object Notation) is a minimal format for semi-structured data used to capture relationships between fields and values.
+
+materialization:
+  displayText: materialization 
+  hoverSnippet: The exact Data Definition Language (DDL) that dbt will use when creating the model’s equivalent in a data warehouse. 
+
+model: 
+  hoverSnippet: A model is an essential building block of the DAG
+  displayText: model
+
+monotonically-increasing:
+  displayText: monotonically increasing 
+  hoverSnippet: A monotonically-increasing sequence is a sequence whose values are sorted in ascending order and do not decrease. For example, the sequences 1, 6, 7, 11, 131 or 2, 5, 5, 5, 6, 10.
+
+predicate-pushdown:
+  displayText: Predicate pushdown
+  hoverSnippet: A predicate pushdown is an expression used to determine what rows in a database apply to a particular query
+
+primary-key:
+  displayText: primary key  
+  hoverSnippet: A primary key is a non-null column in a database object that uniquely identifies each row.
+
+relational-database:
+  displayText: relational database
+  hoverSnippet: A relational database provides a structured way to store data into tables consisting of rows and columns. Different tables in a relational database can be joined together using common columns from each table, forming relationships.
+
+reverse-etl:
+  displayText: reverse ETL
+  hoverSnippet: Reverse ETL is the process of getting your transformed data stored in your data warehouse to end business platforms, such as sales CRMs and ad platforms. 
+
+subquery:
+  displayText: subquery
+  hoverSnippet: A subquery is a query within another query. Subqueries are often used when you need to process data in multiple steps.
+
+surrogate-key:
+  displayText: surrogate key  
+  hoverSnippet: A surrogate key is a unique identifier derived from the data itself. It often takes the form of a hashed value of multiple columns that will create a uniqueness constraint for each row.
+
+table:
+  displayText:  table 
+  hoverSnippet: In simplest terms, a table is the direct storage of data in rows and columns.  Think excel sheet with raw values in each of the cells.  
+
+view:
+  displayText: view  
+  hoverSnippet: A view (as opposed to a table) is a defined passthrough SQL query that can be run against a database (or data warehouse).
+
+# Test terms (DELETE BEFORE MERGE)
+demoTerm:
+  hoverSnippet: This is a demo term hover snippet
+  displayText: Yay demo term
+
+noDisplayText:
```

Comment:
Not sure of the use case here. Also it seems to not be working on the test page, but likely it's that I'm not understanding what it should be doing 😆

Reply:
Yeah feel free to disregard these ones 😅 i just wanted to add some test cases for us to review to ensure the pages won't break in certain scenarios. The `noDisplayText` example falls back to displaying the `id` text itself (`noHoverSnippet`) on the page, if both the `displayText` property and `children` are missing.

Here's an actual use case on this functionality, say we had a term `snowflake` in the `hover-terms.md` file:
```
snowflake:
  hoverSnippet: This is the hover snippet text when hovering over the Snowflake term
```

Then the `Term` component is used like this on a page: `<Term id="snowflake" />`

Since it doesn't have a `displayText` property set in `hover-terms.md`, and child text isn't passed into the component (for example: `<Term id="snowflake">this is child text</Term>`), it will fallback to showing the `id` on the page instead, which is `snowflake`.

Reply:
Ah! Thanks for the explanation!